mac/Keyman4MacIM/Keyman4MacIM/Logging/KMSentryHelper.m

```diff
@@ -0,0 +1,98 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * Created by Shawn Schantz on 2025-01-13.
+ *
+ * Description
+ */
+
+#import "KMSentryHelper.h"
+@import Sentry;
+
+@implementation KMSentryHelper
+
+NSString * const kApiComplianceTagName = @"apiCompliance";
+NSString * const kArchitectureTagName = @"architecture";
+NSString * const kOskVisibleTagName = @"oskVisible";
+NSString * const kClientAppIdTagName = @"clientAppId";
+NSString * const kKeyboardTagName = @"keyboard";
+NSString * const kHasAccessibilityTagName = @"accessibilityEnabled";
+NSString * const kActiveKeyboardCountTagName = @"activeKeyboardCount";
+NSString * const kLifecycleStateTagName = @"lifecycleState";
+
++ (void)addLifecycleStateTag: (NSString *)value {
+  [self addCustomTag:kLifecycleStateTagName withValue:value];
+}
+
++ (void)addApiComplianceTag: (NSString *)value {
+  [self addCustomTag:kApiComplianceTagName withValue:value];
+}
+
++ (void)addArchitectureTag: (NSString *)value {
+  [self addCustomTag:kArchitectureTagName withValue:value];
+}
+
++ (void)addOskVisibleTag:(BOOL)value {
+  [self addCustomTag:kOskVisibleTagName withValue:value?@"true":@"false"];
+}
+
++ (void)addClientAppIdTag:(NSString *)value {
+  [self addCustomTag:kClientAppIdTagName withValue:value];
+}
+
++ (void)addKeyboardTag:(NSString *)value {
+  [self addCustomTag:kKeyboardTagName withValue:value];
+}
+
++ (void)addHasAccessibilityTag:(BOOL)value {
+  [self addCustomTag:kHasAccessibilityTagName withValue:value?@"true":@"false"];
+}
+
++ (void)addActiveKeyboardCountTag:(NSUInteger)value {
+  [self addCustomTag:kActiveKeyboardCountTagName withValue:[[NSNumber numberWithUnsignedLong:value] stringValue]];
+}
+
+/**
+ *assign custom keyboard tag in Sentry to initial keyboard
+ */
++ (void)addCustomTag:(NSString *)tagName withValue:(NSString *)value {
+  [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+    [scope setTagValue:value forKey:tagName];
+  }];
+}
+
++ (void)addBreadCrumb:(NSString *)category message:(NSString *)messageText {
+  SentryBreadcrumb *crumb = [[SentryBreadcrumb alloc] init];
+  crumb.level = kSentryLevelInfo;
+  crumb.category = category;
+  crumb.message = messageText;
+  [SentrySDK addBreadcrumb:crumb];
+}
+
++ (void)addDebugBreadCrumb:(NSString *)category message:(NSString *)messageText {
+  SentryBreadcrumb *crumb = [[SentryBreadcrumb alloc] init];
+  crumb.level = kSentryLevelDebug;
+  crumb.category = category;
+  crumb.message = messageText;
+  [SentrySDK addBreadcrumb:crumb];
+}
+
++ (void)addUserBreadCrumb:(NSString *)category message:(NSString *)messageText {
+  SentryBreadcrumb *crumb = [[SentryBreadcrumb alloc] init];
+  crumb.type = @"user";
+  crumb.level = kSentryLevelInfo;
+  crumb.category = category;
+  crumb.message = messageText;
+  [SentrySDK addBreadcrumb:crumb];
+}
```

Comment:
Would it be cleaner to refactor these into a single function (with a level and an optional type parameters)?

Reply:
I'll take a look at simplifying this. I'm applying an interesting cross section of arguments, and I'd rather have multiple different functions with fewer parameters because that would easier to read in the client code. I should still be able to eliminate duplication in the wrapper at least.

Reply:
I added comments and removed one method not being used, but I think the three functions available are sufficient and would be cleaner from the caller's perspective than creating a variadic function. Also, the duplication is pretty minimal and probably not worth complicating the code to reduce it.